pkgs/development/python-modules/apache-airflow/default.nix

```diff
@@ -102,94 +101,136 @@ buildPythonPackage rec {
-      --replace "flask-appbuilder>=1.12.5, <2.0.0" "flask-appbuilder" \
-      --replace "flask-admin==1.5.3" "flask-admin" \
+      --replace "flask-appbuilder~=2.2" "flask-appbuilder" \
+      --replace "flask-admin==1.5.4" "flask-admin" \
       --replace "flask-login>=0.3, <0.5" "flask-login" \
       --replace "pendulum==1.4.4" "pendulum" \
       --replace "cached_property~=1.5" "cached_property" \
-      --replace "dill>=0.2.2, <0.3" "dill" \
+      --replace "dill>=0.2.2, <0.4" "dill" \
       --replace "configparser>=3.5.0, <3.6.0" "configparser" \
-      --replace "jinja2>=2.10.1, <2.11.0" "jinja2" \
+      --replace "jinja2>=2.10.1, <2.12.0" "jinja2" \
       --replace "colorlog==4.0.2" "colorlog" \
-      --replace "funcsigs==1.0.0" "funcsigs" \
-      --replace "flask-swagger==0.2.13" "flask-swagger" \
-      --replace "python-daemon>=2.1.1, <2.2" "python-daemon" \
+      --replace "funcsigs>=1.0.0, <2.0.0" "funcsigs" \
+      --replace "flask-swagger>=0.2.13, <0.3" "flask-swagger" \
+      --replace "python-daemon>=2.1.1" "python-daemon" \
       --replace "alembic>=1.0, <2.0" "alembic" \
       --replace "markdown>=2.5.2, <3.0" "markdown" \
-      --replace "future>=0.16.0, <0.17" "future" \
+      --replace "future>=0.16.0, <0.19" "future" \
       --replace "tenacity==4.12.0" "tenacity" \
-      --replace "text-unidecode==1.2" "text-unidecode" \
       --replace "tzlocal>=1.4,<2.0.0" "tzlocal" \
       --replace "sqlalchemy~=1.3" "sqlalchemy" \
-      --replace "gunicorn>=19.5.0, <20.0" "gunicorn"
+      --replace "sqlalchemy_jsonfield~=0.9" "sqlalchemy_jsonfield>=0.1" \
+      --replace "gunicorn>=19.5.0, <21.0" "gunicorn" \
+      --replace "werkzeug<1.0.0" "werkzeug" \
+      --replace "importlib_resources~=1.4" "importlib_resources" \
+      --replace "importlib-metadata~=2.0" "importlib-metadata>=1.0" \
+      --replace "marshmallow-sqlalchemy>=0.16.1, <0.24.0" "marshmallow-sqlalchemy" \
+      --replace "requests>=2.20.0, <2.24.0" "requests>=2.20.0" \
+      --replace "lazy_object_proxy<1.5.0" "lazy-object-proxy"
+    # NOTE: preference above is to relax the constraint completely, but when the constraint
+    # is of format ``package>1.0.0; python_version<"3.7"``, we have to put ANY bound
 
-    # dumb-init is only needed for CI and Docker, not relevant for NixOS.
-    substituteInPlace setup.py \
-      --replace "'dumb-init>=1.2.2'," ""
+    # fix issues w/ upgrade to pendulum >= 2.0
+    substituteInPlace airflow/settings.py \
+      --replace "from pendulum import Pendulum" "from pendulum import DateTime as Pendulum"
+    substituteInPlace tests/core/test_core.py \
+      --replace "from pendulum import utcnow" "import pendulum; import functools; utcnow = functools.partial(pendulum.now, 'UTC')"
 
-    substituteInPlace tests/core.py \
-      --replace "/bin/bash" "${stdenv.shell}"
+    # --replace "from pendulum import utcnow" "from datetime import datetime as _dt; utcnow = _dt.utcnow"
```

Comment:
I think it would be better to provide a second, separate, `pendulum_1` derivation on `pythonPackages`, and put this patch behind an overridable option... there are lots of bits of Airflow that import pendulum so using pendulum 2 might be the wrong approach here, see https://github.com/apache/airflow/search?q=pendulum

Here's a working override that can point to what's needed for a pendulum_1 derivation:

```nix
        pendulum = pythonSuper.pendulum.overridePythonAttrs (oldAttrs: rec {
          inherit (oldAttrs) pname;
          version = "1.4.4";
          name = "${pname}-${version}";
          src = fetchPypi {
            inherit pname version;
            sha256 = "0p5c7klnfjw8f63x258rzbhnl1p0hn83lqdnhhblps950k5m47k0";
          };

          propagatedBuildInputs = oldAttrs.propagatedBuildInputs
            ++ (with pythonSelf; [ tzlocal ]);
        });
```